sw/device/tests/pmp_smoketest_napot.c

```diff
@@ -54,6 +57,12 @@ static void pmp_configure_load_napot(void) {
 OTTF_DEFINE_TEST_CONFIG();
 
 bool test_main(void) {
+  // Check that `PMP_LOAD_REGION_ID` region is not already used.
+  uint32_t pmpcfg1 = 0;
+  CSR_READ(CSR_REG_PMPCFG1, &pmpcfg1);
+  CHECK((pmpcfg1 & 0xff000000) == 0,
```

Comment:
Nit: I'm not sure I like that `0xff000000` is hard-coded here but is tied to `PMP_LOAD_REGION_ID` being `7`. I understand it would be annoying to select the register and byte offset based on this constant, though.

If you don't want to calculate it, could you add a `static_assert(PMP_LOAD_REGION_ID == 7)` or `&&` that into the `CHECK` so it always stays up to date?

Reply:
I added a `pmp_region_is_configured` function to `runtime/pmp`

Reply:
Very nice, thanks.

Would you mind sticking this in `sram_ctrl_execution_test` while we're here?

Reply:
The current pmp API doesn't seem to be able to handle the napot usage in `sram_ctrl_execution_test`. `0x7fffffff` napot means that `2**34` bytes and this overflows the 32-bit integer API.